src/server/pfs/server/master.go

```diff
@@ -58,11 +58,42 @@ func (d *driver) master(ctx context.Context) {
 }
 
 func (d *driver) finishCommits(ctx context.Context) error {
-	compactor, err := newCompactor(d.env.BackgroundContext, d.storage, d.etcdClient, d.prefix, d.env.StorageConfig.StorageCompactionMaxFanIn)
+	repos := make(map[string]context.CancelFunc)
```

Comment:
Is this missing a deferred for loop to cancel all of these?

Reply:
I was leaning on the context always being canceled if we exit this function (context canceled elsewhere or returning an error canceling the error group context). It is probably not good to rely on that though, so I can add a deferred cancel to ensure those contexts are canceled.